backends/sqlite/sqlite_backend_test.go

```diff
@@ -0,0 +1,837 @@
+	}
+
+	db, err = sql.Open("sqlite3", dbPath)
+	if err != nil {
+		t.Fatalf("unable to open database: %s", err.Error())
+	}
+
+	// Delete everything in the neoq_jobs table if it exists
+	_, _ = db.Exec("DELETE FROM neoq_jobs")
+
+	// Delete everything in the neoq_dead_jobs table if it exists
+	_, _ = db.Exec("DELETE FROM neoq_dead_jobs")
+
+	return dbURL, db
+}
+
+// TestBasicJobProcessing tests that the sqlite backend is able to process the most basic jobs with the
+// most basic configuration.
+func TestBasicJobProcessing(t *testing.T) {
+	connString, db := prepareAndCleanupDB(t)
+	const queue = "testing"
+	maxRetries := 5
+	done := make(chan bool)
+	defer close(done)
+
+	timeoutTimer := time.After(5 * time.Second)
+
+	ctx := context.Background()
+	nq, err := neoq.New(ctx,
+		neoq.WithBackend(sqlite.Backend),
+		sqlite.WithConnectionString(connString),
+		neoq.WithLogLevel(logging.LogLevelDebug),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer nq.Shutdown(ctx)
+
+	h := handler.New(queue, func(_ context.Context) (err error) {
+		done <- true
+		return
+	})
+
+	err = nq.Start(ctx, h)
+	if err != nil {
+		t.Error(err)
+	}
+
+	deadline := time.Now().UTC().Add(5 * time.Second)
+	jid, e := nq.Enqueue(ctx, &jobs.Job{
+		Queue: queue,
+		Payload: map[string]interface{}{
+			"message": "hello world",
+		},
+		Deadline:   &deadline,
+		MaxRetries: &maxRetries,
+	})
+
+	if e != nil || jid == jobs.DuplicateJobID {
+		t.Error(e)
+	}
+
+	select {
+	case <-timeoutTimer:
+		err = jobs.ErrJobTimeout
+	case <-done:
+	}
+	if err != nil {
+		t.Error(err)
+	}
+
+	// ensure job has fields set correctly
+	var jdl time.Time
+	var jmxrt int
+	var jqueue string
+
+	err = db.
+		QueryRow("SELECT queue, deadline,max_retries FROM neoq_jobs WHERE id = $1", jid).
+		Scan(&jqueue, &jdl, &jmxrt)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if jqueue != queue {
+		t.Error(fmt.Errorf("job queue does not match its expected value: %v != %v", jqueue, queue))
+	}
+
+	jdl = jdl.In(time.UTC)
+	// dates from postgres come out with only 6 decimal places of millisecond precision, naively format dates as
+	// strings for comparison reasons. Ref https://www.postgresql.org/docs/current/datatype-datetime.html
+	if jdl.Format(time.RFC3339) != deadline.Format(time.RFC3339) {
+		t.Error(fmt.Errorf("job deadline does not match its expected value: %v != %v", jdl, deadline)) // nolint: goerr113
+	}
+
+	if jmxrt != maxRetries {
+		t.Error(fmt.Errorf("job MaxRetries does not match its expected value: %v != %v", jmxrt, maxRetries)) // nolint: goerr113
+	}
+}
+
+func TestMultipleProcessors(t *testing.T) {
+	const queue = "testing"
+	var execCount uint32
+	var wg sync.WaitGroup
+
+	connString, _ := prepareAndCleanupDB(t)
+
+	h := handler.New(queue, func(_ context.Context) (err error) {
+		atomic.AddUint32(&execCount, 1)
+		wg.Done()
+		return
+	})
+	// Make sure that each neoq worker only works on one thing at a time.
+	h.Concurrency = 1
```

Comment:
This possibly places the test under unrealistic conditions. I would expect real world users to use default concurrency. Will this test work with concurrency > 1?

Reply:
These are test cases borrowed from postgres backend. I hadn't put much thought on this specific case. I can try your suggestion.